### PR TITLE
Aeon hem gen5 association fix and Configuration update

### PR DESCRIPTION
**Pull request by Aeotec-ccheng**
- Group Association fix present under "def Refresh()"

- def configuration() parameter settings updated for device setting optimization (kWh and Watt report every 300 seconds)

### CODE REVIEW

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (greens) [inline devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:160]: I think we'd want to keep this, since it guarantees we get consistent settings on configure.

Unless you can confirm whether or not settings are preserved on removing and re-adding the device to a network.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (greens) [inline devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:162]: so this reports power and energy on one association group instead of two, right?

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:160]: All default parameter settings will revert back to their default values upon removing and re-adding the device to a network, there are no persistent parameter settings.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:162]: That is correct, but this isn't really based on Association Groups, both Parameter 101, 102, and 103 all send to Group Association Group #1 specifically.

Each group just allows you to send various sensors of the HEM Gen5 at different time intervals. (as Group1 interval, Group2 Interval, Group3 Interval)_

Since originally 101 was set to watt, and 102 was set to kWh, but both were adjusted to report at 300, this is essentially the same as setting both kWh + Watt to report on 101 only.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (kianooshkarami) [inline devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:143]: I am kind of curious as to reasons that prompted this change? The hub automatically does this upon join for the past 4-5 years that I can remember. Anything that you'd like the hub team to look into?

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (kianooshkarami) [inline devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:167]: Could we potentially leverage `CONFIGURATION_BULK_SET` command to generate one command instead of one per each configuration?

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (cbaumler) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:143]: I'm also curious about this.

Comment (greens) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:143]: my guess is that it's because it's not explicitly shown anywhere that we do this

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (greens) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:160]: great, then this is fine.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (greens) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:167]: I think maybe we'd want to do it ourselves first in another DTH before asking a third party to.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:143]: I assume that Association Group 1 is set to Node ID 01 automatically, but i have seen and gotten many reported cases where Group 1 is not successfully associated to Node ID 01 where the HEM Gen5 is not reporting automatically.

I have had to help large amount of supports where adjusting Group 01 to report to NodeID 01 using Z-Wave Tweaker custom device handler has allowed the HEM Gen5 to report normally,  which adding these 2 lines of code ensures that the HEM Gen5 is properly assigned without having to use a separate device handler.

---

I have noticed that Z-Wave controllers (not just SmartThings hub) that are using MULTICHANNEL ASSOCIATION to assign Group 1 to Node ID 1.1 which causes this issue, as Node ID 1.1 is not responded to by HEM Gen5 (HEM Gen5 uses an earlier form of Multichannel Association). For this to work, Node ID 1.0 for the use of this command class to work.

This code line is more of a bandage assuming that the association settings are automated and would reduce the amount of reported issues where the HEM Gen5 is not reporting without refreshing the page manually.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:167]: Unfortunately HEM Gen5 does not respond to CONFIGURATION BULK SET and BULK GET, please see PC Controller 5 logs below

Capture

**Notes -** The last 2 reports by HEM Gen5 are issued via CONFIGURATION GET 0x65 and CONFIGURATION GET 0x6F individually

----

Parameter 102 and 103 settings can be removed since it typically wouldn't really have any affect on the HEM Gen5 reports as a possible way to reduce configuration lines.

Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:167]: Parameter 102 and 103 settings can be removed since it typically wouldn't really have any affect on the HEM Gen5 reports as a possible way to reduce configuration lines.

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (cbaumler) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:143]: I don't think having this bandage in the DTH hurts anything, but I verified that the SmartThings hub adds node ID 1 / endpoint ID 0 to group 1 (lifeline) if the device supports Multichannel Association V3. So it should work. I don't have a device to verify with, but I'd be more than willing to investigate if you have customers running into this on our hub.

**Review by greens — APPROVED** [review]
lgtm, but I'll wait for one other sign-off before I merge

**Review by cbaumler — APPROVED** [review]

**Review by cjswedes — APPROVED** [review]

**Comment by greens** [comment]
@Aeotec-ccheng this will be live in production on the 1st

File: devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeon-home-energy-meter.src/aeon-home-energy-meter.groovy:143]: @cbaumler - It should be working based on that information. If i do notice more of these in the future, i'll start debugging the backend to see if there are any problems via ZSniffer as well during its pair process.

**Pull request merged**
